dsa-solutions/lc-solutions/0200-0299/0206-Reverse-Linkedlist.md

```diff
@@ -0,0 +1,299 @@
+          while node:
+              print(node.val, end=" ")
+              node = node.next
+
+      # Driver code
+      if __name__ == "__main__":
+          solution = Solution()
+          head = ListNode(1)
+          head.next = ListNode(2)
+          head.next.next = ListNode(3)
+          head.next.next.next = ListNode(4)
+          head.next.next.next.next = ListNode(5)
+
+          print("Original list: ", end="")
+          printList(head)
+
+          head = solution.reverseList(head)
+
+          print("\nReversed list: ", end="")
+          printList(head)
+      ```
+    </TabItem>
+    <TabItem value="c" label="C">
+      <SolutionAuthor name="@Ajay-Dhangar"/>
+      ```c
+      #include <stdio.h>
+      #include <stdlib.h>
+
+      // Definition for singly-linked list.
+      struct ListNode {
+          int val;
+          struct ListNode *next;
+      };
+
+      void solve(struct ListNode** head, struct ListNode* prev, struct ListNode* curr) {
+          if (curr == NULL) {
+              *head = prev;
+              return;
+          }
+          struct ListNode* nextNode = curr->next;
+          solve(head, curr, nextNode);
+          curr->next = prev;
+      }
+
+      struct ListNode* reverseList(struct ListNode* head) {
+          if (head == NULL || head->next == NULL) {
+              return head;
+          }
+          solve(&head, NULL, head);
+          return head;
+      }
+
+      // Helper function to print the linked list
+      void printList(struct ListNode* node) {
+          while (node != NULL) {
+              printf("%d ", node->val);
+              node = node->next;
+          }
+      }
+
+      // Driver code
+      int main() {
+          struct ListNode* head = (struct ListNode*)malloc(sizeof(struct ListNode));
+          head->val = 1;
+          head->next = (struct ListNode*)malloc(sizeof(struct ListNode));
+          head->next->val = 2;
+          head->next->next = (struct ListNode*)malloc(sizeof(struct ListNode));
+          head->next->next->val = 3;
+          head->next->next->next = (struct ListNode*)malloc(sizeof(struct ListNode));
+          head->next->next->next->val = 4;
+          head->next->next->next->next = (struct ListNode*)malloc(sizeof(struct ListNode));
+          head->next->next->next->next->val = 5;
+          head->next->next->next->next->next = NULL;
+
+          printf("Original list: ");
+          printList(head);
+
+          head = reverseList(head);
+
+          printf("\nReversed list: ");
+          printList(head);
+
+          return 0;
+      }
+      ```
+    </TabItem>
+
+</Tabs>
+
+## Video lecture
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/D2vI2DNJGd8?si=fCV_x4Mwv--u4yUb" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
```

Comment:
### Bad Practice

```md
<iframe width="560" height="315" src="https://www.youtube.com/embed/D2vI2DNJGd8?si=fCV_x4Mwv--u4yUb" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
```
### Good Practice

```md
<LiteYouTubeEmbed
    id="VIDEO_ID_HERE"
    params="autoplay=1&autohide=1&showinfo=0&rel=0"
    title="VIDEO_TITLE_HERE"
    poster="maxresdefault"
    webp />

    <br />
```

Reply:
for Example

```jsx
<LiteYouTubeEmbed
    id="lI3RBnK8V6Y"
    params="autoplay=1&autohide=1&showinfo=0&rel=0"
    title="CodeHarborHub Website Introduction | CodeHarborHub Website Overview | CodeHarborHub Website Features"
    poster="maxresdefault"
    webp />
```

and Output is

https://codeharborhub.github.io/docs/#what-is-codeharborhub

Reply:
what i will write in id params and poster

Reply:
oh I understood